contrib/docker/README.md

```diff
@@ -0,0 +1,108 @@
+# Synapse Docker
+
+This Docker image will run Synapse as a single process. It does not provide any
+database server or TURN server that you should run separately.
+
+If you run a Postgres server, you should simply have it in the same Compose
+project or set the proper environment variables and the image will automatically
+use that server.
+
+## Build
+
+Build the docker image with the `docker build` command from the root of the synapse repository.
+
+```
+docker build -t matrixdotorg/synapse:v0.22.1 .
+```
+
+The `-t` option sets the image tag. Official images are tagged `matrixdotorg/synapse:<version>` where `<version>` is the same as the release tag in the synapse git repository.
+
+You may have a local Python wheel cache available, in which case copy the relevant packages in the ``cache/`` directory at the root of the project.
+
+## Run
+
+It is recommended that you use Docker Compose to run your containers, including
+this image and a Postgres server. A sample ``docker-compose.yml`` is provided,
+including example labels for reverse proxying and other artifacts.
+
+Then, to run the server:
+
+```
+docker-compose up -d
+```
+
+In the case you specified a custom path for you configuration file and wish to
+generate a fresh ``homeserver.yaml``, simply run:
+
+```
+docker-compose run --rm synapse generate
+```
+
+If you do not wish to use Compose, you may still run this image using plain
+Docker commands. Note that the following is just a guideline and you may need
+to add parameters to the docker run command to account for the network situation
+with your postgres database.
+
+```
+docker run \
+    -d \
+    --name synapse \
+    -v ${DATA_PATH}:/data \
+    -e SYNAPSE_SERVER_NAME=my.matrix.host \
+    matrixdotorg/synapse:v0.22.1
```

Comment:
This should probably be pointed at master, otherwise the version will be forgotten about and a bunch of old servers will be spun up.

Reply:
Pointing at ``master`` sounds a bit risky as well. Maybe point to ``latest``, that the team will have a responsibility to point to the current working version?

Reply:
master is effectively latest, but yes: latest would be safer.

Reply:
Oh yes, I had forgotten about that regarding this repo. But sure, let's go for ``latest`` anyway.